docs/reference/async-processing.mdx

```diff
@@ -0,0 +1,116 @@
+
+1. The greater the remote latency, the bigger the impact of async. At 2 ms round trip latency, throughput improves by 5.5x. At 1 ms round trip latency, throughput improves by 3x.
+2. The throughput with async processing enabled is stable even with varying latency.
+
+:::note
+The async processor makes the round trip network latency a non-factor in achieving high throughput. However, using the async processor will generally place more load on the remote system, which in turn may get saturated, which in turns will drive up the latencies, which will bring down the throughput gains of the async processor. 
+
+Even so, the async processor performed at a steady 30% better even when the remote system was at its limit in our testing.
+:::
+
+We are continuing to test and tune the async processor, and will share more results as we have them. 
+
+## Enabling async processing
+
+Turning async processing on (or off!) is quick and easy. All it requires is one line of code and one config and you’re good to go! Make sure to check out the “Limitations” section for what is covered by the current version and what will be coming in a future release.
+
+### 1. Select and update your existing Processor
+
+Wrap the supplier for the processor you wish to enable async processing for in the appropriate wrapper class:
+
+    1. For a `.process()` with a `ProcessorSupplier`, use `AsyncProcessorSupplier#createAsyncProcessorSupplier`
+    2. For a `.processValues()` with a  `FixedKeyProcessorSupplier` , use `AsyncFixedKeyProcessorSupplier#createAsyncProcessorSupplier`
+    
+Example:
+    
+```java
+    // Before
+    kstream.processValues(
+        new MyStatefulProcessorSupplier<>(),
+        Named.as("RegularProcessor"),
+        STORE_NAME);
+    
+    // After
+    kstream.processValues(
+        createAsyncProcessorSupplier(
+            new MyStatefulProcessorSupplier<>(),
+            Named.as("AsyncProcessor"),
+            STORE_NAME);
+            
+    // note: the Named.as was modified to highlight the example,
+    // changing the processor name is not required to enable async processing
+```
+    
+### 2. Configure the thread pool
+
+Configure the thread pool size by setting the ResponsiveConfig `responsive.async.thread.pool.size` to a positive integer. Something like 10 is a good place to start, but make sure to check out the tuning section below for a more detailed discussion of how to tune and configure the Responsive Async Processor.
+
+### 3. Ensure that you implement the Processor#stores method
+
+Note that the async processing framework requires a stateful processor to connect its StateStores by implementing the `ProcessorSupplier#stores` method. Please verify that your processor supplier implements this method before passing it into the async wrapper, or switch to this method of connecting state stores if it does not.
```

Comment:
ah I just saw the note in limitations.

Reply:
yeah it's just a silly artificial requirement due to how we connect things internally right now --we can and absolutely should fix this in the future, it's just non-trivial so it was put on the backburner since this feature is most sorely needed by stateful apps anyway. But good question (should we maybe say something here too, or add a note to see the "Limitations" section for an explanation on this?